style/react.md

```diff
@@ -1,440 +1,18 @@
-    state: { hi: number }
-
-    // React lifecycle hooks.
-    // They should follow their chronological ordering:
-    // 1. componentWillMount
-    // 2. componentDidMount
-    // 3. componentWillReceiveProps
-    // 4. shouldComponentUpdate
-    // 5. componentWillUpdate
-    // 6. componentDidUpdate
-    // 7. componentWillUnmount
-    componentDidMount() { ... }
-
-    // All other Flow annotations
-    props: { ... }
-    someRandomData: string,
+> Follow the normal [JavaScript style guide](javascript.md). We generally try to follow the official [React tutorials and guides](https://react.dev/reference/react). Please default to them as a source-of-truth.
 
-    // All other instance methods
-    handleClick = (e) => { ... }
-
-    // Finally, the render method
-    render() { ... }
-}
-```
-
-#### Name handlers `handleEventName`.
-
-Example:
-
-```jsx
-<Component onClick={this.handleClick} onLaunchMissiles={this.handleLaunchMissiles} />
-```
-
-#### Name handlers in props `onEventName`.
-
-This is consistent with React's event naming: `onClick`, `onDrag`,
-`onChange`, etc.
-
-Example:
-
-```jsx
-<Component onLaunchMissiles={this.handleLaunchMissiles} />
-```
-
-
-#### Open elements on the same line.
-
-The 80-character line limit is a bit tight, so we opt to conserve the extra 4.
-
-Yes:
-```jsx
-return <div>
-   ...
-</div>;
-```
-
-No:
-```jsx
-return (      // "div" is not on the same line as "return"
-    <div>
-        ...
-    </div>
-);
-```
-
-#### Align and sort HTML properties.
-
-Fit them all on the same line if you can. If you can't, put each
-property on a line of its own, indented four spaces, in sorted order.
-The closing angle brace should be on a line of its own, indented the
-same as the opening angle brace. This makes it easy to see the props
-at a glance.
-
-Yes:
-```jsx
-<div className="highlight" key="highlight-div">
-<div
-    className="highlight"
-    key="highlight-div"
->
-<Image
-    className="highlight"
-    key="highlight-div"
-/>
-```
+----
 
-No:
-```jsx
-<div className="highlight"      // property not on its own line
-     key="highlight-div"
->
-<div                            // closing brace not on its own line
-    className="highlight"
-    key="highlight-div">
-<div                            // property is not sorted
-    key="highlight-div"
-    className="highlight"
->
-```
+## Syntax
 
-#### Only export a single react class.
+### Prefer to export a single React component
 
-Every .jsx file should export a single React class, and nothing else.
-This is for testability; the fixture framework requires it to
-function.
+Every .tsx file should ideally export a single React component.
+This is for testability and composition; it's easier to write Storybook stories and
+it's clearer where the breaks in functionality are.
 
-Note that the file can still define multiple classes, it just can't export
+Note that the file can still define multiple components, it just shouldn't export
```

Comment:
Worth a comment here to encourage defining separate components at the module level?  Rather than internally?  As I understand it, this help React better decide what to rerender.

Reply:
Oh we should definitely callout against creating components inside of components as that causes all sorts of rendering issues. One can memoize to reduce that, but really, components generated in components that also render them is a pattern we should avoid as it always leads to rendering problems like infinite or long render loops.